extensions/blocks/simple-payments/index.js

```diff
@@ -189,5 +190,5 @@ export const settings = {
 		reusable: false,
 	},
 
-	deprecated: [ deprecatedV1 ],
+	deprecated: [ deprecatedV1, deprecatedV2 ],
```

Comment:
Not blocking this PR or anything, but I was always wondering if it would work to async-load these deprecations. They're basically doubling/tripling the block JS+CSS size each time.

cc @sgomes for thoughts

Reply:
If at all possible, I would heartily recommend that, yes. Looking at the code, however, it's not immediately obvious how to do that easily.

Perhaps making `deprecated` an async function to an array, rather than a simple array?

```js
{
  deprecated: async () => {
    const deprecatedV1 = await import( './depecrated/v1' );
    const deprecatedV2 = await import( './depecrated/v2' );
    return [ deprecatedV1, deprecatedV2 ];
  },
}
```

Other code that makes use of these properties would then have to be updated accordingly, and made to be async-safe.

Fixing the `depecrated` typo would also be nice 😄

Reply:
Yeah, so maybe this is something to add support for in core first. cc @mtias @youknowriad

Reply:
Conversation to be continued in #17467.